frontend/src/elements/components/ShareData.js

```diff
@@ -0,0 +1,282 @@
+        >
+          Save
+        </button>
+      </form>
+    );
+  }
+}
+
+const Delete = ({ onDelete, onCancel }) => (
+  <React.Fragment>
+    <p>Are you sure you want to delete this item? This action cannot be rolled back.</p>
+    <div style={{ marginTop: 20 }}>
+      <button onClick={onCancel} className="button">
+        Go back
+      </button>
+      &nbsp;
+      <button onClick={onDelete} className="button is-danger">
+        {"I'm sure"}
+      </button>
+    </div>
+  </React.Fragment>
+);
+
+class SharedData extends React.Component {
+  state = {
+    shared: {}
+  };
+
+  componentDidMount() {
+    window.cg.Subject.getSharedData(this.props.sharedDataToken)
+      .then(shared => {
+        this.setState({ shared });
+      })
+      .catch(err => {
+        console.log('failure', err);
+      });
+  }
+
+  render() {
+    return (
+      <ul style={{ fontSize: 14 }}>
+        {Object.keys(this.state.shared).map((k, i) => (
+          <li key={i}>
+            {k}: {this.state.shared[k]}
+          </li>
+        ))}
+      </ul>
+    );
+  }
+}
+
+class ShareDataItem extends React.Component {
+  state = {
+    isEdit: false,
+    isDelete: false
+  };
+
+  showOnDelete = e => {
+    e.preventDefault();
+    this.setState({ isDelete: true });
+  };
+
+  showOnEdit = e => {
+    e.preventDefault();
+    this.setState({ isEdit: true });
+  };
+
+  render() {
+    const { share, onUpdate, onDelete, onCopy } = this.props;
+    const { isEdit, isDelete } = this.state;
+
+    return (
+      <div className={`${styles.card} ${(isEdit || isDelete) && styles.cardHover}`}>
+        {isEdit ? (
+          <div className={styles.cardBack}>
+            <EditForm
+              initialValues={share}
+              onSubmit={onUpdate}
+              onCancel={e => {
+                e.preventDefault();
+                this.setState({ isEdit: false });
+              }}
+            />
+          </div>
+        ) : isDelete ? (
+          <div className={`${styles.cardBack} ${styles.cardDelete}`}>
+            <Delete
+              onDelete={e => onDelete(e, share.id)}
+              onCancel={e => {
+                e.preventDefault();
+                this.setState({ isDelete: false });
+              }}
+            />
+          </div>
+        ) : (
+          <React.Fragment>
+            <a className={`is-pulled-right ${styles.deleteIcon}`} onClick={this.showOnDelete}>
+              <span className="icon">
+                <i className="far fa-trash-alt" />
+              </span>
+            </a>
+            <div>
+              <span className={styles.shareImg}>
+                <img src={clipboardSvg} alt={share.name} style={{ height: 35 }} />
+              </span>
+              <div className={styles.shareName}>{share.name}</div>
+              <div className={styles.shareUrl}>{share.url}</div>
+              <SharedData sharedDataToken={share.token} />
+              {/* <button
```

Comment:
Edit doesn't make sense, but I added that since it was in the design first, and then check that perhaps don't make to much sense, at least Rectification has something to do with it.